src/ray/raylet/worker_pool.cc

```diff
@@ -24,7 +24,7 @@ std::shared_ptr<ray::raylet::Worker> GetWorker(
 
 // A helper function to remove a worker from a list. Returns true if the worker
 // was found and removed.
-bool RemoveWorker(std::list<std::shared_ptr<ray::raylet::Worker>> &worker_pool,
+bool RemoveWorker(std::vector<std::shared_ptr<ray::raylet::Worker>> &worker_pool,
```

Comment:
Removes get a bit slower, but we're already doing an O(n) find anyways so it doesn't really matter.

Reply:
why not just use an `unordered_set` for workers then, average case O(1) for insert, erase, and size. I don't think we care about worker order.